pkg/messages/baseMessage.go

```diff
@@ -55,6 +57,12 @@ type RequestMessage interface {
 	Reply(senderFloor int, response bool) ResponseMessage
 }
 
+type ProposalMessage interface {
+	Message
+	Treaty() Treaty
+	Reply(senderFloor int, response bool) ResponseMessage
```

Comment:
```suggestion
```

Do we need a response if we're just making a proposal?

Reply:
Yes...?

Reply:
But it's a proposal - we're not replying to anything

Reply:
The recipient of `ProposalMessage` calls `Reply()` to respond to the proposal

Reply:
Ah ok, my lack of understanding. I agree then